### PR TITLE
qiime2-2019.10

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

File: qp_qiime2/qp_qiime2.py [hunk 1]
```diff
@@ -377,9 +377,9 @@ def call_qiime2(qclient, job_id, parameters, out_dir):
         try:
             q2params['reads'] = qiime2.Artifact.import_data(
                 'FeatureData[Sequence]', fna_fp)
-        except ValueError as e:
+        except (ValueError, qiime2.core.exceptions.ValidationError) as e:
             msg = str(e)
-            if 'Invalid characters in sequence' in msg:
+            if 'Invalid characters on line' in msg:
                 msg = ('Table IDs are not sequences, please confirm that this '
                        'is not a close reference table?')
```
Comment (ElDeveloper) [inline qp_qiime2/qp_qiime2.py]: This should be closed not close, right?

File: qp_qiime2/__init__.py [hunk 2]
```diff
@@ -21,6 +21,20 @@
 from qiime2.sdk import PluginManager
 
 
+def _get_qiime2_type_name_and_predicate(element):
```
Comment (ElDeveloper) [inline qp_qiime2/__init__.py]: Any way to have some tests for this function? I also am not sure I follow 100% what is going on. It seems like Union types have to be handled different but I don't understand what the first member is the representative member of the element.

**Review by ElDeveloper — CHANGES_REQUESTED** [review]
Looks good, although adding some tests might be beneficial!

File: qp_qiime2/util.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    """helper method to get the qiime2 type name and predicate
+
+    Parameters
+    ----------
+    element : qiime2.core.type.signature
+        The signature to parse
+
+    Returns
+    -------
+    name, predicate
```
Comment (ElDeveloper) [inline qp_qiime2/util.py]:
```suggestion
    str, str
```

File: qp_qiime2/util.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        The name and the predite of the signature passed
```
Comment (ElDeveloper) [inline qp_qiime2/util.py]:
```suggestion
        The name and the predicate of the inputed signature
```

File: qp_qiime2/util.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    """
+    to_ast = element.qiime_type.to_ast()
+    if to_ast['type'] == 'union':
+        # union types allow to give choises to another type of paramter; for
```
Comment (ElDeveloper) [inline qp_qiime2/util.py]:
```suggestion
        # union types allow to give choices to another type of paramter; for
```

File: qp_qiime2/util.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        # example for a range(1, 10) give the choise `ignore`. These are not
```
Comment (ElDeveloper) [inline qp_qiime2/util.py]:
```suggestion
        # example for a range(1, 10) give the choice `ignore`. These are not
```

**Review by ElDeveloper — CHANGES_REQUESTED** [review]
Thanks for adding tests. I've added a few minor changes but should be good to go otherwise.

**Comment by antgonza** [comment]
Thank you soooo much!

**Comment by ElDeveloper** [comment]
👍

**Pull request merged**